Assets/Scripts/Behaviours/Vehicles/Vehicle.cs

```diff
@@ -416,7 +417,18 @@ private void Update()
             {
                 UpdateColors();
             }
-            
+
+            if (currentRadioStation != 0 && Ped.Instance.CurrentVehicle == this)
```

Comment:
`Ped.Instance` can be null.

`Ped.Instance` must be the driver.

Reply:
why driver? passenger cant listen to radio?

Reply:
Yes, passenger should hear radio. But, this way, when ped enters a vehicle as pessenger, `ContinueRadio()` will be called which will switch the current song. Am I right ?

Reply:
Currently radio station is not synced between players so I don't see a problem here.

Reply:
No, it switches to the next song always, because `radio.isPlaying` will be false when ped exits the vehicle. Correct me if I am wrong.

Reply:
Yes, it will switch song when ped (driver or passenger) enters vehicle. I guess radio should be started in `VehicleSittingState.EnterVehicleInternal` then?

Reply:
Ok, you can do it from there. But make sure to restore previous time of the song. Don't restart the song every time when ped enters the vehicle.